pvanalytics/quality/time.py

```diff
@@ -33,3 +36,130 @@ def spacing(times, freq):
     # timestamp is considered valid.
     delta.iloc[0] = freq
     return delta == freq
+
+
+def _round_multiple(x, to, up_from=None):
+    # Return `x` rounded to the nearest multiple of `to`
+    #
+    # Parameters
+    # ----------
+    # x : Series
+    # to : int
+    #     `x` is rounded to a multiple of `to`
+    # up_from : int, optional
+    #     If the remainder of `x` / `to` is greater than `up_from`
+    #     then `x` is rounded up, otherwise `x` is rounded down.
+    #     If not specified rounding will go up from `to` // 2.
+    up_from = up_from or to // 2
+    quotient, remainder = divmod(abs(x), to)
+    remainder[remainder > up_from] = to
+    remainder[remainder <= up_from] = 0
+    return np.sign(x) * (quotient*to + remainder)
+
+
+def shifts_ruptures(daytime, clearsky_midday, period_min=2,
+                    shift_min=15, round_up_from=None,
+                    prediction_penalty=13):
+    """Identify time shifts using the ruptures library.
+
+    Compares the solar-transit time from `clearsky_midday` with the
+    mid-point between sunrise and sunset time derived from the `daytime`
+    mask. These times are the times of the first and last True value on
+    each day.
+
+    The Pelt changepoint detection method is applied to the difference
+    in midday times derived from the `daytime` series and the expected
+    midday times from `clearsky_midday`. For each period between change
+    points the mode of the difference is rounded to a multiple of
+    `shift_min` and returned as the time-shift for all values in that
+    period.
+
+    Parameters
+    ----------
+    daytime : Series
+        Boolean time-series with True for day and False for night.
+    clearsky_midday : Series
```

Comment:
Thinking if this input would be better named `transit` for the solar transit time, or local noon. It doesn't have much to do with clearsky conditions, it's just the midpoint between sunrise and sunset.